s3/s3.js

```diff
@@ -0,0 +1,68 @@
+const AWS = require('aws-sdk');
+const { logger } = require('../services');
+
+let s3 = new AWS.S3({ endpoint: 'http://localstack:4572', region: 'us-west-1', s3ForcePathStyle: true });
+
+
+function createBucket(params) {
+  const result = s3.createBucket(params).promise()
+    .then(data => {
+      logger.debug(data);
+      logger.info(`Successful creation of S3 bucket ${params.Bucket}`);
+      return true;
+    })
+    .catch(err => {
+      logger.error(`Error creating S3 bucket ${params.Bucket} \nError: ${err}`);
+      return false;
+    });
+
+  return result;
+}
+
+function bucketExists(params) {
+  const result = s3.headBucket(params).promise()
+    .then(data => {
+      logger.debug(`Bucket ${params.Bucket} exists`);
+      return true;
+    })
+    .catch(err => {
+      logger.debug(`Bucket ${params.Bucket} does not exist`);
+      return false;
+    });
+
+  return result;
+}
+
+function listBuckets() {
+  const result = s3.listBuckets().promise()
+    .then(data => {
+      logger.debug(`Successfully retrieved bucket list:\n${data}`);
+      return data;
+    })
+    .catch(err => {
+      logger.error(`Could not retreive bucket list. \n Error: ${err}`);
+    });
+
+  return result;
+}
+
+function putObject(params) {
+  const result = s3.putObject(params).promise()
+    .then(data => {
+      logger.info(`Successful write to S3 key: ${params.Key}`);
+      logger.debug(`Successful write to S3 key: ${params.Key}\n Payload: ${params.payload}`);
+      return data;
+    })
+    .catch(err => {
+      logger.error(`Error writing data to S3 target: ${err}`);
```

Comment:
Should throw err or resolve `null` as per the docs.

Reply:
Wrote an errorHandler to take care of this and throw the error.